src/common/pick_address.cc

```diff
@@ -352,6 +352,10 @@ int pick_addresses(
 	  (ipv & CEPH_PICK_ADDRESS_IPV6)) {
 	r = fill_in_one_address(cct, ifa, CEPH_PICK_ADDRESS_IPV6, networks,
 				interfaces, addrs, preferred_numa_node);
+
+        if (r >= 0) {
```

Comment:
By jumping out the loop here, you are breaking the pick_address logic when we want both ipv4 and ipv6 address and ipv4 is not the preferred type.

Reply:
Yeah, I think that my colleague and I didn't understand what exact behavior was truly expected. Thanks very much for your comments!